### DISCUSSION TITLE
Cannot export river package as shapefile properly

### DISCUSSION THREAD

**Discussion by sebmeng** [modflowpy/flopy#2309] [Q&A] [answered]
Hello community:
I have some troubles to export the river package of a model I created. If I use the function `riv.export(os.path.join(model_ws, "riv.shp"))` I get a shapefile of the whole model, but the only attributes that it has is the node number.

If I use `flopy.plot.PlotMapView` I get a nice view of the cells that has the river boundary condition:

```
#plot river BC using flopy
fig, ax = plt.subplots(1, 1, figsize=(10, 8), constrained_layout=True)
mv = flopy.plot.PlotMapView(model=gwf, ax=ax, layer=6) #in layer 7 the bottom is present
mv.plot_bc("RIV",plotAll=True)
```
![image](https://github.com/user-attachments/assets/2a293c96-76b3-4e2d-a852-ca787feef3c2)

and if I export the model to a VTK file and add the river package, I can also see them quite easily: 

![image](https://github.com/user-attachments/assets/701b4b5b-953c-4056-a4ea-abb95cea1cf8)

But I need them in shp format to make some checking with GIS. Maybe I am doing something wrong in the export process?

Thanks in advance for your help!
Cheers,
Sebastian

**Comment by martclanor** [comment, marked as answer]
Hi Sebastian,

Trying to look around here. Do you maybe happen to have multiple stress periods in your model while your river package's stress period data is defined only on a subset of your stress periods, (perhaps the initial one)?

If it's anything similar to that, then my guess is that it's because the `MFPandasTransientList` does not re-use the data from the preceding stress periods when it is empty (which is probably intended as mf6 would handle that logic already). In that case, an error is caught on the following code block, and thus exporting of the stress period data is skipped.

https://github.com/modflowpy/flopy/blob/181e101a605bdb9b628c6781abff7b3276aca635/flopy/export/shapefile_utils.py#L380-L383

I can try to submit a PR on this, but maybe the faster solution for you is to just assign the river stress period data for each stress period.

Regards,
Mart

**Reply by sebmeng** [reply]
Hi Mart,
That is indeed how I managed the river data, since MF6 copies the information of the previous defined stress-period I just have the information of the first SP defined in my RIV package. I will add the info for the rest of the SP and see how it goes,
Thanks!

Sebastian
